src/boundary.js

```diff
@@ -0,0 +1,105 @@
+import constant from "./constant";
+
+export default function(x0, y0, x1, y1) {
+  var strength = constant(0.1),
+      hardBoundary = true,
+      border = constant( Math.min((x1 - x0)/2, (y1 - y0)/2) ),
+      nodes,
+      strengthsX,
+      strengthsY,
+      x0z, x1z,
+      y0z, y1z,
+      borderz,
+      halfX, halfY;
+
+
+  if (typeof x0 !== "function") x0 = constant(x0 == null ? -100 : +x0);
+  if (typeof x1 !== "function") x1 = constant(x1 == null ? 100 : +x1);
+  if (typeof y0 !== "function") y0 = constant(y0 == null ? -100 : +y0);
+  if (typeof y1 !== "function") y1 = constant(y1 == null ? 100 : +y1);
+
+  function getVx(halfX, x, strengthX, border, alpha) {
+    return (halfX - x) *  Math.min(2, Math.abs( halfX - x) / halfX) * strengthX * alpha;
+  }
+
+  function force(alpha) {
+    for (var i = 0, n = nodes.length, node; i < n; ++i) {
+      node = nodes[i];
+
+      if ((node.x < (x0z[i] + borderz[i]) || node.x > (x1z[i] - borderz[i])) ||
+          (node.y < (y0z[i] + borderz[i]) || node.y > (y1z[i] - borderz[i])) ) {
+        node.vx += getVx(halfX[i], node.x, strengthsX[i], borderz[i], alpha);
+        node.vy += getVx(halfY[i], node.y, strengthsY[i], borderz[i], alpha);
+      } else {
+        node.vx = 0;
+        node.vy = 0;
```

Comment:
I tried this and found that setting vx/vy to zero broke my simulation (i'm applying forceBoundry last).
most likely this `else` should be removed

Reply:
this may be what @hornj encountered

Reply:
Thanks @kumavis, I'll check it out